resource/src/main/java/io/smallrye/common/resource/PathResource.java

```diff
@@ -0,0 +1,107 @@
+package io.smallrye.common.resource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.Charset;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.FileTime;
+import java.time.Instant;
+
+import com.sun.nio.file.ExtendedOpenOption;
+
+import io.smallrye.common.constraint.Assert;
+
+/**
+ * A resource corresponding to a {@link Path}.
+ */
+public final class PathResource extends Resource {
+    private static final long MAP_THRESHOLD = 256 << 10; // 256KiB
```

Comment:
Any particular reason for choosing this threshold?

Reply:
No, it's 100% gut feeling. That said, I'm a bit nervous about mapping resources on an open-ended basis. I think I might revisit the API so that the lifetime of the buffer is bounded. This way on Java 21+ we can actively unload the buffer once the call is complete. I'll have to think about it a little bit.